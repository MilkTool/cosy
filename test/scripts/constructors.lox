
class Dog {
    init(text) {
        this.text = text
        //return nil // early returns are possible
        //print "unreachable"
    }
    say() {
        print this.text + ", bark bark!"
    }
}

var dog = Dog("I'm a dog")
dog.say()

print dog.init("I'm a cat in disguise") // reinitialization is allowed, returning the object
dog.say()

class Sheep {
    // no constructor
    say() {
        print "Bah!"
    }
}
var sheep = Sheep()
sheep.say()
//sheep.init() // error
