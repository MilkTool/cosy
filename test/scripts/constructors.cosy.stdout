I'm a dog, bark bark!
Dog instance
I'm a cat in disguise, bark bark!
Bah!
